broker/plugins/kubejobs/plugin.py

```diff
@@ -233,6 +235,59 @@ def errors(self):
             return ()
         return self.rds.lrange("job:errors", 0, -1)
 
+    def validate(self, data):
+        data_model = {
+            "cmd": list,
+            "control_parameters": dict,
+            "control_plugin": six.string_types,
+            "env_vars": dict,
+            "img": six.string_types,
+            "init_size": int,
+            "monitor_info": dict,
+            "monitor_plugin": six.string_types,
+            "redis_workload": six.string_types,
+            "enable_visualizer": bool
+            # The parameters below are only needed if enable_visualizer is True
+            # "visualizer_plugin": six.string_types
+            # "visualizer_info":dict
+        }
+
+        for key in data_model:
+            if (key not in data):
+                raise ex.BadRequestException(
+                    "Variable \"{}\" is missing".format(key))
+            if (not isinstance(data[key], data_model[key])):
+                raise ex.BadRequestException(
+                    "\"{}\" has unexpected variable type: {}. Was expecting {}"
+                    .format(key, type(data[key]), data_model[key]))
+
+        if (data["enable_visualizer"]):
+            if ("visualizer_plugin" not in data):
+                raise ex.BadRequestException(
+                    "Variable \"visualizer_plugin\" is missing")
+
+            if (not isinstance(data["visualizer_plugin"], six.string_types)):
+                raise ex.BadRequestException(
+                    "\"visualizer_plugin\" has unexpected variable type: {}.\
+                     Was expecting {}"
+                    .format(type(data["visualizer_plugin"]),
+                            data_model["visualizer_plugin"]))
```

Comment:
there's no such key in data_model :-)

Reply:
This is because this parameter is optional, thats why it's  commented in the data model, but they do are required if the enable visualizer is set.